segmenters/audio/AudioSlicer/__init__.py

```diff
@@ -24,15 +23,17 @@ def _segment(self, signal):
         import librosa
         if signal.ndim == 1:  # mono
             frames = librosa.util.frame(signal, frame_length=self.frame_length, hop_length=self.hop_length, axis=0)
+            return frames,
+
         elif signal.ndim == 2:  # stereo
             left_frames = librosa.util.frame(
                 signal[0,], frame_length=self.frame_length, hop_length=self.hop_length, axis=0)
             right_frames = librosa.util.frame(
                 signal[1,], frame_length=self.frame_length, hop_length=self.hop_length, axis=0)
-            frames = np.concatenate((left_frames, right_frames), axis=0)
+            return left_frames, right_frames
```

Comment:
what is the reasoning behind this change?

Reply:
I've got the same question here. The only difference is to convert the return from a `ndarray` to a tuple of `ndarray`

Reply:
At the end, both left and right frames are considered the same chunks.

Reply:
It differs in how segmented chunks are stored and reconstructed later if needed.

For `mono` it is same as before
`[ {'offset':` 0, 'location': [0, 10]}, {'offset': 1, 'location': [10, 20]}, {'offset': 2, 'location': [20, 30]} ]`

But for stereo it is different. For each channel in the audio offset starts from 0. This is done keeping reconstruction in mind.
When you see `offset=0` for the second time while traversing the list, you know the chunks for the second channel has started
`[ {'offset': 0, 'location': [0, 10]}, {'offset': 1, 'location': [10, 20]}, {'offset': 2, 'location': [20, 30]},
    {'offset': 0, 'location': [0, 10]}, {'offset': 1, 'location': [10, 20]}, {'offset': 2, 'location': [20, 30]} ]`
    
So when concatenated array is returned from the `_segment` there is no way to understand where the second channel frames start from because we do not have length information. That is why returning a tuple of frames in both cases of mono and stereo and looping over them. Thus no length information is required.

Reply:
@nan-wang @JoanFM what do you think?

Reply:
@shakurshams Yes, I see your point. However, when you assign the chunks at L53, the frames from the right channel will override those from the left.

Reply:
@nan-wang I have pasted the output from the last two test cases(for one signal instead of two for brevity.) For stereo, chunks from both channels are preserved following the scheme described above. Please let me know if I missed your point.
**Stereo**
`[[{'offset': 0, 'weight': 1.0, 'blob': array([ 0.52708615, -0.21221454, -1.0650824 , -0.48549198, -0.08167043,
       -0.38034069,  1.84052205,  1.26871832, -0.683173  ,  0.21671767]), 'location': [0, 10]}, {'offset': 1, 'weight': 1.0, 'blob': array([-1.55060346, -2.26714673,  0.63778872,  0.86921508,  1.43830099,
       -1.75240552,  2.09746676,  0.77696722,  1.74378927,  0.2115113 ]), 'location': [10, 20]}, {'offset': 2, 'weight': 1.0, 'blob': array([-1.65390314, -0.04860204,  0.03966604, -0.20333415,  1.57110702,
        2.7834961 ,  0.51260664,  0.89482025,  0.10671443,  0.69157241]), 'location': [20, 30]}, {'offset': 3, 'weight': 1.0, 'blob': array([ 0.01885564, -0.8876197 ,  0.67363343, -0.37729471, -0.129511  ,
       -0.93622562, -0.16900971, -1.07119272,  0.85680062, -2.29911036]), 'location': [30, 40]}, {'offset': 4, 'weight': 1.0, 'blob': array([ 0.35992368, -0.04810679, -2.1906127 ,  1.06116999,  0.78800577,
        0.62752204,  0.86104441, -1.46333466,  2.11331119,  1.66678649]), 'location': [40, 50]}, {'offset': 0, 'weight': 1.0, 'blob': array([ 0.23663368, -2.04901163, -1.26061622,  0.00849126, -0.80775948,
        0.32778837,  0.47677929,  0.61443827, -0.83340377, -0.0171386 ]), 'location': [0, 10]}, {'offset': 1, 'weight': 1.0, 'blob': array([-0.39726652,  0.40275055,  1.06923626,  0.46275345, -2.01034733,
       -0.72624474,  0.7360374 , -0.52888659,  0.93563908,  0.88169216]), 'location': [10, 20]}, {'offset': 2, 'weight': 1.0, 'blob': array([ 1.92583051, -1.96131975,  0.9781112 , -0.70289831, -0.46250962,
        1.1666676 ,  0.69058627,  1.53404274,  0.28370558,  0.66857541]), 'location': [20, 30]}, {'offset': 3, 'weight': 1.0, 'blob': array([-1.18176242,  0.87834701,  0.83459   , -2.63591185, -0.57192406,
       -0.88206868,  0.79404908,  1.1386616 , -0.46092391, -0.05199123]), 'location': [30, 40]}, {'offset': 4, 'weight': 1.0, 'blob': array([ 1.06539497, -1.33079596,  0.89450982,  1.9121258 ,  0.14376057,
       -0.35467981, -0.37416159, -1.54486295, -1.27437986,  0.08780342]), 'location': [40, 50]}]]`
       
**Mono:**
`[[{'offset': 0, 'weight': 1.0, 'blob': array([-0.45984967, -0.63631383, -0.66508786, -0.06480296, -1.65522152,
       -0.63912864,  0.14825386,  0.72792373,  0.22836782,  0.03122948]), 'location': [0, 10]}, {'offset': 1, 'weight': 1.0, 'blob': array([-0.3267693 , -0.23741972, -1.30259229, -1.12866133, -0.07516026,
        0.7373738 , -1.63273306, -0.65924093,  0.11204031,  1.0211533 ]), 'location': [10, 20]}, {'offset': 2, 'weight': 1.0, 'blob': array([-0.75463424,  0.13538528,  1.25098552,  0.83004809, -0.31403187,
        0.57925337, -0.79246062, -1.25144412,  1.35942729, -0.4936783 ]), 'location': [20, 30]}, {'offset': 3, 'weight': 1.0, 'blob': array([ 0.49459769, -0.23233242, -1.56765216, -0.57442509,  0.81551088,
       -1.00448579, -0.94726661, -0.67283012, -0.71853731, -1.54388408]), 'location': [30, 40]}, {'offset': 4, 'weight': 1.0, 'blob': array([ 0.18335447,  0.36723736,  1.50778129,  0.02037247, -1.12726425,
       -0.52474678,  1.95070634,  0.21581817, -0.17478254,  0.55028923]), 'location': [40, 50]}]]`